### PR TITLE
Remove taxons

### PR DESCRIPTION
**Pull request by mbj**
* Just to get a sense what it takes to kill a frontend only feature

### CODE REVIEW

File: api/app/controllers/spree/api/taxonomies_controller.rb [hunk 1]
```diff
@@ -1,64 +0,0 @@
...
-      def index
-        respond_with(taxonomies)
-      end
-
-      def show
-        respond_with(taxonomy)
-      end
-
-      # Because JSTree wants parameters in a *slightly* different format
-      def jstree
```
Comment (dkubb) [inline api/app/controllers/spree/api/taxonomies_controller.rb]: @mbj is it possible that jstree only existed to support taxonomies? it appears so to me:

```bash
$ ag -i jstree
backend/app/assets/javascripts/spree/backend.js
10://= require jquery.jstree/jquery.jstree

backend/app/assets/stylesheets/spree/backend/plugins/_jstree.scss
1:#vakata-dragged.jstree-apple .jstree-invalid,
2:#vakata-dragged.jstree-apple .jstree-ok,
3:#jstree-marker {
8:#vakata-dragged.jstree-apple .jstree-invalid {
13:#vakata-dragged.jstree-apple .jstree-ok {
19:#jstree-marker {
25:#jstree-marker-line {

backend/app/assets/stylesheets/spree/backend/spree_admin.scss
27:@import 'spree/backend/plugins/jstree';

backend/lib/spree/backend/engine.rb
23:          jquery.jstree/themes/apple/*
```

**Comment by dkubb** [comment]
@mbj I would support removing taxons. There's one outstanding question about jstree in this PR.

Also I assume there would be migrations to remove all taxon related tables? I think you mentioned this already, but just confirming in a place where we won't forget.

**Comment by mbj** [comment]
> Also I assume there would be migrations to remove all taxon related tables? I think you mentioned this already, but just confirming in a place where we won't forget.

Yeah, I'm about to kill these after we did the schema collapse as the first migration on top of the collapse.

File: api/app/controllers/spree/api/taxonomies_controller.rb (hunk 1 above)
Comment (mbj) [inline api/app/controllers/spree/api/taxonomies_controller.rb]: Good catch, I sucessfully expanded into removing jstree.

**Comment by mbj** [comment]
@dkubb This is ready for review. I did not yet remove the schema, lets do that on the next import from MRH. Removing the code reduces liability already (and improves spec runtime).

**Comment by dkubb** [comment]
@mbj it looks like there's one reference to "taxon" in `api/config/reek.yml`. I'll remove this now and merge.

**Comment by mbj** [comment]
> @mbj it looks like there's one reference to "taxon" in api/config/reek.yml. I'll remove this now and merge.

thx.

**Pull request merged**